src/dictionary.c

```diff
@@ -198,6 +198,27 @@ static void SpellCheckDictAuxSave(RedisModuleIO *rdb, int when) {
   dictReleaseIterator(iter);
 }
 
+static void SpellCheckDictAuxSave2(RedisModuleIO *rdb, int when) {
+  if (when == REDISMODULE_AUX_BEFORE_RDB) {
+    return;
+  }
+  if (dictSize(spellCheckDicts)) {
```

Comment:
We need to decide how to handle empty suggestion dictionary.

if we consider an empty suggestion dictionary as non-existing (there is no command to delete it, just to delete its entries) and we avoid serializing it to rdb (need to change `aux_save`) could it be considered as non-breaking?
And if only empty suggestion dictionaries exist, rdb will not have any related data, and the search module would not be needed to load such an rdb.

It seems the change in behavior is is that we will create the Redis key not when loading the rdb (since it will not be there), but only when calling `FT.SUGADD` on such empty dictionaries.

Before the change, we would create the empty Redis key already when loading from rdb.

If we keep writing empty suggestion dictionaries, we would still need the search module to load rdbs which has such dictionaries, which is not benefiting from the intention of `aux_save2`, so maybe an alternative could be to skip writing empty suggestion dictionaries if all of them are empty, and to write them if at least one of them is non-empty. Then the search module would be required to load an rdb only if is contains actual related data (meaning that suggestion dictionary names on their own are insignificant)

Reply:
Update, some changes related to spell check dictionaries:
* We are not loading empty dictionaries, we skip them.
* We are deleting the dictionary when it becomes empty.
* Due to the previous changes, there are no empty dictionaries when we are saving the rdb.